src/sentry/sentry_metrics/indexer/cache.py

```diff
@@ -140,9 +157,18 @@ def bulk_record(
             return cache_key_results
 
         db_record_key_results = self.indexer.bulk_record(use_case_id, db_record_keys.mapping)
+
+        # We will temporarily change it to write new cache keys for all results
+
+        # DB key results
         self.cache.set_many(
             db_record_key_results.get_mapped_key_strings_to_ints(), use_case_id.value
         )
+
+        self.cache.set_many_new(
+            db_record_key_results.get_mapped_key_strings_to_ints(), use_case_id.value
+        )
+
```

Comment:
Looks like the migration to the new namespace will only happen to the cache misses (db reads) because the indexer only calls cache set_many on db reads. I suppose we're okay with this instead of something like
```python
def bulk_record(
        self, use_case_id: UseCaseKey, org_strings: Mapping[int, Set[str]]
    ) -> KeyResults:
    res = self._old_bulk_record(use_case_id, org_strings)
    self.cache.set_many_new(res.get_mapped_key_strings_to_ints(), use_case_id.value)
    return res
```
Because the cache is evicted in 2 hours, no matter what?

Reply:
Yep exactly. If it works today (where keys are constantly being evicted and re-created) then this should also work, is my understanding.